easier_net/easier_net.py

```diff
@@ -0,0 +1,174 @@
+from typing import List
+import numpy as np
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from torch.utils.data import TensorDataset
+from torch.utils.data import DataLoader
+
+
+import pickle
+import sier_net 
+
+
+class EasierNetEstimator:
+    """ This is the Ensembling by Averaging Sparse-Input Hierarchical Networks(EASIER-NET)"""
+    def __init__(
+        self,
+        num_classes=0, #Equal to 0 if doing a regression, otherwise number of classes for binary / multi-classification
+        input_filter_layer=True, #Scales the inputs by parameter β
+        n_layers=2, #Number of hidden layers
+        n_hidden=10, #Number of hidden nodes in each layer
+        n_estimators=5, #Size of ensemble; number of SIER-nets being ensembled.
+        max_iters=40, #Number of epochs to run Adam     
+        max_prox_iters=0, #Number of epochs to run batch proximal gradient descent
+        input_pen=0, #λ1; Controls the input sparsity         
+        full_tree_pen=0.001, #λ2; controls the number of active layers and hidden nodes 
+        num_batches=3, #number of mini-batches for Adam
+        model_fit_params_file=None, #A json file that specifies what the hyperparameters are. If given, this will override the arguments passed in.
+    ):
+
+        self.num_classes=num_classes
+        self.input_filter_layer=input_filter_layer
+        self.n_layers=n_layers
+        self.n_hidden=n_hidden
+        self.n_estimators=n_estimators
+        self.max_iters=max_iters
+        self.max_prox_iters=max_prox_iters
+        self.full_tree_pen=full_tree_pen
+        self.input_pen=input_pen
+        self.num_batches=num_batches
+        self.model_fit_params_file=model_fit_params_file
+        self.score_criterion = (
+            nn.CrossEntropyLoss() if self.num_classes >= 2 else nn.MSELoss()
+        )       
+        if self.model_fit_params_file is not None:
+            self.estimators: List[sier_net.SierNetEstimator] = self.load_model(model_fit_params_file) 
+        else:
+            self.estimators: List[sier_net.SierNetEstimator] = self._generate_estimators()
+
+        assert num_classes != 1
+        assert input_filter_layer
+    
+    def _generate_estimators(self):
+        estimators=[]
+        for est in range(self.n_estimators): #TODO: i -> est?
+            est=sier_net.SierNetEstimator(
```

Comment:
how come `batch_size` is missing?

Reply:
the argument `n_layers` also seems to be missing. can you check if there are other arguments missing here?